### ISSUE TITLE
Phantom blocks

### ISSUE THREAD

**Issue by DrLeonardo** [Team-RTG/Realistic-Terrain-Generation#759]
Very often on the map there are " phantom blocks" . Areas such units are enormous , when you try to move to the area player throws after rejoining the server problem is solved . But not for long. How to solve this problem ?

The problem is seen in the version of RTG-1.7.10-0.8.0-snapshot- 5 , but it is very unobtrusive , on versions of the above problem is critical

KCouldron 1.7.10 1614 201 (checked on Thermos 56)
![hmcnsutm5om](https://cloud.githubusercontent.com/assets/18382530/16157901/4517323c-34c3-11e6-91a6-3ca63f3e078f.jpg)
![upswjd7pypi](https://cloud.githubusercontent.com/assets/18382530/16157899/45106dda-34c3-11e6-9cf6-1efb9a3c8299.jpg)
![wioyubtpba8](https://cloud.githubusercontent.com/assets/18382530/16157900/45123cbe-34c3-11e6-8b1e-d3335bb195eb.jpg)

**Comment by srs-bsns** [comment]
I haven't noticed a single instance of anything like this, nor has anyone else reported anything like this. I assume this is a KCouldron issue.

**Comment by DrLeonardo** [comment]
Tests have shown that the core server is not guilty .
Now there is a bug in the Thermos

**Comment by DrLeonardo** [comment]
The problem is seen in the version of RTG-1.7.10-0.8.0-snapshot- 5 , but it is very unobtrusive , on versions of the above problem is critical

**Comment by whichonespink44** [comment]
@DrLeonardo We don't officially support KCauldron, so there''s not much we can do to help. The only way we could help is if you could reproduce this issue using only a Forge server.

The one thing I could suggest is to try using the latest 1.0 snapshot to see if it fixes the problem. Just be sure to backup your world because upgrading from 0.8-snapshot-5 *might* create some chunk walls.

I'm closing this issue, but I'll re-open if it turns out to be reproducible using only Forge.

**Comment by ghost** [comment]
@whichonespink44 I can confirm this. I using 1.0-snapshot 2 will happend when I only use Forge in single game. Also KCauldron had this problem too.
**And now 90% of Mod servers are using KCauldron or Thermos, you said do not support KCauldron, I think this is not a wise choice.**

**Comment by DrLeonardo** [comment]
@sourcepowered I agree , do support KCauldron and Thermos

**Comment by whichonespink44** [comment]
So this is happening with just Forge and RTG installed? If so, please provide a seed so we can investigate further.

**Comment by ghost** [comment]
@whichonespink44 OK, sir. I will try to reproduce the problem.

**Comment by ghost** [comment]
@whichonespink44 
I find the problem. Please see:
Used in Forge Client, single game, verson 1.7.10-1.0.1.
![123](https://cloud.githubusercontent.com/assets/17511838/16539617/441ccb9e-407b-11e6-8fd9-db758eca8278.png)
And you can see this area is extremely unusual.
![123123](https://cloud.githubusercontent.com/assets/17511838/16539621/6dfcacfe-407b-11e6-90a2-5ac4fb98463f.png)

**And I also tried to use the KCauldron and Thermos, it's still not normal.
The log file is very normal, no any warning or error.**

And, you can see that problem is a regular distribution (Looks like a rectangle):
![123123123](https://cloud.githubusercontent.com/assets/17511838/16539649/2acd5130-407c-11e6-9575-d9d3ee9af381.png)

**Comment by whichonespink44** [comment]
Please post the entire log file (fml-client-latest.log) and I'll try to reproduce the problem.

**Comment by ghost** [comment]
@whichonespink44 The trees does not seem to be some ghost blocks ... But the shrub still has the problem. And I will send you the world archive and log file, please wait.
You can see this, it's not normal I think (I will send you this archive):
![123111](https://cloud.githubusercontent.com/assets/17511838/16539705/a5f6db50-407d-11e6-8e25-f54fba559475.png)
**The coordinates are: -213, 51.**

**It seems that RTG is very supportive of KCauldron and Thermos, which should be Mod's own BUG.**

**Comment by ghost** [comment]
@whichonespink44 
The world archive: https://mega.nz/#!MUBWmRqb!KgBcDI3CeB96dyPg48ZX3QZBXAbFwmgIyRusPSsUqac

The log file (It's too big, I can't upload it to pastebin.com): https://mega.nz/#!QVQGXTDZ!RESLLQMlL8F5bJy5rWHUnOdOdP5jCG9fLoIgx_YuRFE

**Comment by whichonespink44** [comment]
Ok, thanks for that stuff.

We'll take a look in a couple hours and see if we can figure out what's going on.

**Comment by ghost** [comment]
@whichonespink44 
What the ... Please see this:
![125123121](https://cloud.githubusercontent.com/assets/17511838/16539849/e329f594-4081-11e6-88b0-c6530c524f83.png)
Is still just archive I send you, and the coordinates are -241, 220.
I only once again load the world ..........

**Comment by ghost** [comment]
@whichonespink44 
After the actual test, **if I am in the client to generate a world, then I'll copy the world to the server,** will also produce the above problems: **The world chasm** ?!

**Even modifying the "level-tyle" to "RTG" is invalid.**

**Comment by ghost** [comment]
And, RTG does not seem to support the mutil world:
First, generate a normal world.
Second, use the Multiverse plugin to import a already exists RTG world.
Third, **the world chasm**???!!! Almost as above.

This has nothing to do with KCauldron or Thermos, it should be the problem of Mod itself I think.
So I wonder if it's the second time loading the map that will lead to the problem.

---

**I'm sorry I said so much, but now there are 3 problems:
1. Ghost blocks.
2. The world chasm?!
3. Can not support Mutil world (even modify the level-type).**

**Comment by srs-bsns** [comment]
> Can not support Mutil world (even modify the level-type).

RTG doesn't work outside of Dim 0.

**Comment by ghost** [comment]
@srs-bsns @whichonespink44 

> RTG doesn't work outside of Dim 0.

But RWG, BOP and other Mods are supported.
I don't think we should maverick, support the Thermos and KCauldron not a difficult thing. And the number of 1.7.10 users currently still more and more ( I think many people don't like Microsoft, 1.8, you know ) and Thermos and KCauldron is the best server side. BOP said no longer support 1.7.10, and finally they are updated because too many users using 1.7.10.
We need your help. If you do not support, we can only use the old version (such as RWG) of mod, if there is a BUG or a lot of bugs, we can not fix it, because we are not a programmer. We can only let the server go to worse.
Please help 1.7.10 users, like me, like him. Thank you very much, as well as a server creator.

**Comment by topisani** [comment]
if you are still using 1.7.10, i know the solution to your problem.
start stopping.

yes, everything else is kinda in flux right now, but modders are settling down on 1.9.4 currently, and we too will soon have a usable version out.

its impossible for us to keep maintaining a mod for a version of minecraft that is approaching an age of 3 years, while staying up to date with the newest versions at the same time, let alone fix some weird bug in an api that we do not support.

No one should be starting new servers in 1.7.10 now anyway, and if you are using kcauldron and have these issues, you probably have already stopped using rtg, as the incompatibility looks pretty severe.

**Comment by whichonespink44** [comment]
@sourcepowered The log file that you provided is showing that you have a lot of other mods installed in addition to RTG. Please try reproducing this issue with just Forge and RTG installed. And be sure to provide a world seed so we can investigate further. In the meantime, I'm going to close this issue again, as it's almost certainly nothing to do with RTG.

**Comment by ghost** [comment]
@topisani 
Yes, sir, you are right, but there are some big problems:
If quit using Bukkit, choose what server software and it must have PLUGINS function?
Is it really only use mojang's server software with forge?
You know, if there is no plugins, you can not protect the server, to achieve more functionality, such as login(like AuthMe), and Plots(like PlotSquared) and some plugins like WordGuard and more and more.
And if there is no plug-in, how anti-cheat? Disable some items(You know some of the items are dangerous, can destroy the server and players' efforts)?
There are many problems simply can not be expressed in words. There are many excellent Mod such as IC2, TF, but they only support 1.7.10, either. (1.8 and 1.9 has Sponge, it's great but it only support 1.8+).
And if using 1.7.10, you can only choose KCauldron or Thermos...............
So, would you please tell me how should I do? Give up all this?
It is obvious that all this is a contradiction.
Sir, I think we must support it, although I am a Spigot user. Sponge in the efficiency and ease of use, compared with Thermos, HUGE DIFFERENCE. we can not compared them, Sponge is too weak, simply not worth mentioning. And Sponge lack of too many important plug-ins. Now most Mod servers are using Thermos.
**The stakes we must know.**

**Comment by whichonespink44** [comment]
@AuthMe You're free to use whatever you want... but we're also free to support (or in this case, not support) whatever _we_ want. And currently, the only server platform we support is Forge.

The reason we only support Forge isn't because we don't like Bukkit or Spigot.... it's because we're a small team with a lot of tasks to do. We just don't have the time to support multiple server platforms at the moment, sorry.

If you absolutely _must_ have realistic terrain, you'll either have to find another mod that does it, or fork RTG and add support yourself.

**Comment by topisani** [comment]
@AuthMe 
1. sponge is the plugin system we have chosen to support, because it is very unobtrusive and generally avoids problems like this - no, its not yet perfect, but evidentally, neither are the things discussed here.  
2. Otherwise, forge alone could be considered - there are mods like FTB-Utils which allow for land claiming and such, but yes, i am aware these things are all way better supported in actual plugin systems.

Other than that, as pink already said, we simply cant support every obscure plugin api out there _especially_ not for a version as old as 1.7.10, and _especially_ not now, two weeks after we have released the final major version for 1.7.10, but if any of you know java, you are free to fix it and send us a pull request.
